mipengine/controller/__init__.py

```diff
@@ -23,3 +26,14 @@ class DeploymentType(str, Enum):
 else:
     with open_text(controller, "config.toml") as fp:
         config = AttrDict(envtoml.load(fp))
+
+
+def _get_algorithms_specifications() -> Dict[str, AlgorithmSpecification]:
+    return {
+        algo_name: algorithm.get_specification()
+        for algo_name, algorithm in algorithm_classes.items()
+        if algorithm.get_specification().enabled
+    }
+
+
+algorithms_specifications = _get_algorithms_specifications()
```

Comment:
Why is this in `controller/__init__`? I think it should be in `algorithm_specification.py`

Reply:
The `algorithm_specification.py` contains the logic that is needed from the algorithms and has no dependency on anything else.

It could be added in the future in the `algorithms` folder but currently there is a problem with that due to being imported multiple times.

The code that you mention has dependencies on many other things, so I think the distinction should be there.

Reply:
As discussed in the call it's better to have this logic separate. Maybe in the future the instantiations could be moved out of the `__init__.py` files.